featuretools/entityset/entityset.py

```diff
@@ -1034,14 +1044,74 @@ def add_last_time_indexes(self, updated_dataframes=None):
                             lti_df['last_time_old'] = lti_df['last_time_old'].astype('datetime64[ns]')
                             lti_df = lti_df.fillna(pd.to_datetime('1800-01-01 00:00')).max(axis=1)
                             lti_df = lti_df.replace(pd.to_datetime('1800-01-01 00:00'), pd.NaT)
-                    # lti_df = lti_df.apply(lambda x: x.dropna().max(), axis=1)
 
-                    dataframe.ww.metadata['last_time_index'] = lti_df
-                    dataframe.ww.metadata.get('last_time_index').name = 'last_time'
+                    es_lti_dict[dataframe.ww.name] = lti_df
+                    es_lti_dict[dataframe.ww.name].name = 'last_time'
 
             explored.add(dataframe.ww.name)
-        self.reset_data_description()
 
+        # Store the last time index on the DataFrames
+        dfs_to_update = {}
+        for df in self.dataframes:
+            lti = es_lti_dict[df.ww.name]
+            if lti is not None:
+                if self.time_type == 'numeric':
+                    # --> don't totally understand why this is necessary when we can do the conversion normally?
+                    # --> need to make sure this works for dask and koalas
+                    if lti.dtype == 'datetime64[ns]':
+                        lti = lti.apply(lambda x: x.value)
+                    lti = ww.init_series(lti, logical_type='Double')
```

Comment:
I think we've always stored the last time index as Datetimes, but now that we're putting it into the DataFrame, I was thinking it should match the entityset's time type. Happy to remove these lines if it's not necessary, though.